Dockerfile

```
@@ -19,28 +23,96 @@ COPY ./app/img/ /source/app/img/
 COPY ./app/ts/ /source/app/ts/
 RUN npm run build
 
+# --------------------------------------------------------
+# Base Image: Create the base image that will host the app
+# --------------------------------------------------------
+
 # Cache the kubo image
 FROM ipfs/kubo:v0.25.0@sha256:0c17b91cab8ada485f253e204236b712d0965f3d463cb5b60639ddd2291e7c52 as ipfs-kubo
 
 # Create the base image
-FROM debian:12.2-slim@sha256:93ff361288a7c365614a5791efa3633ce4224542afb6b53a1790330a8e52fc7d
+FROM debian:12.2-slim@sha256:93ff361288a7c365614a5791efa3633ce4224542afb6b53a1790330a8e52fc7d as base
+
+# Add curl to the base image (7.88.1-10+deb12u5)
+# Add jq to the base image (1.6-2.1)
+RUN apt-get update && apt-get install -y curl=7.88.1-10+deb12u5 jq=1.6-2.1
 
 # Install kubo and initialize ipfs
 COPY --from=ipfs-kubo /usr/local/bin/ipfs /usr/local/bin/ipfs
 
+# Copy app's build output and initialize IPFS api
+COPY --from=builder /source/app /export
 RUN ipfs init
+RUN ipfs add --cid-version 1 --quieter --only-hash -r /export > ipfs_hash.txt
 
-# Copy lunaria build output
-COPY --from=builder /source/app /export
+# --------------------------------------------------------
+# Publish Script: Option to host app locally or on nft.storage
+# --------------------------------------------------------
+
+COPY <<'EOF' /entrypoint.sh
```

Recommend setting the working directory since this script will create some files along the way and it would be good to be explicit about where they end up.  I think `~` will work, but you could set it to literally any directory including one you make up like `/workspace` which is a common choice.
```suggestion
WORKDIR ~
COPY <<'EOF' /entrypoint.sh
```